src/core/components/hv-screen/index.js

```diff
@@ -408,205 +295,27 @@ export default class HvScreen extends React.Component {
-          newRoot = this.doc;
-        }
-        newRoot = Behaviors.setIndicatorsAfterLoad(showIndicatorIdList, hideIndicatorIdList, newRoot);
-        // Re-render the modifications
-        this.setState({
-          doc: newRoot,
-        });
-
-        if (typeof onEnd === 'function') {
-          onEnd();
+    this.props.onUpdate(href, action, currentElement, opts, {
+      clearElementError: () => {
+        if (this.state.elementError) {
+          this.setState({ elementError: null });
         }
-      });
-
-    if (delay) {
-      /**
-       * Delayed behaviors will only trigger after a given amount of time.
-       * During that time, the DOM may change and the triggering element may no longer
-       * be in the document. When that happens, we don't want to trigger the behavior after the time
-       * elapses. To track this, we store the timeout id (generated by setTimeout) on the triggering
-       * element, and then look it up in the document after the elapsed time. If the timeout id is not
-       * present, we update the indicators but don't execute the behavior.
-       */
-      const delayMs = parseInt(delay, 10);
-      let timeoutId = null;
-      timeoutId = setTimeout(() => {
-        // Check the current doc for an element with the same timeout ID
-        const timeoutElement = getElementByTimeoutId(this.doc, timeoutId.toString());
-        if (timeoutElement) {
-          // Element with the same ID exists, we can execute the behavior
-          removeTimeoutId(timeoutElement);
-          fetchAndUpdate();
-        } else {
-          // Element with the same ID does not exist, we don't execute the behavior and undo the indicators.
-          newRoot = Behaviors.setIndicatorsAfterLoad(showIndicatorIdList, hideIndicatorIdList, this.doc);
-          this.setState({
-            doc: newRoot,
-          });
-          if (typeof onEnd === 'function') {
-            onEnd();
-          }
-        }
-      }, delayMs);
-      // Store the timeout ID
-      setTimeoutId(currentElement, timeoutId.toString());
-    } else {
-      // If there's no delay, fetch immediately and update the doc when done.
-      fetchAndUpdate();
-    }
-  }
-
-  /**
-   * Used internally to update the state of things like select forms.
-   */
-  onSwap = (currentElement, newElement) => {
-    const parentElement = currentElement.parentNode;
-    parentElement.replaceChild(newElement, currentElement);
-    const newRoot = shallowCloneToRoot(parentElement);
-    this.setState({
-      doc: newRoot,
+      },
+      getDoc: () => this.doc,
+      getNavigation: () => this.navigation,
+      getOnUpdate: () => this.onUpdate,
+      getStateDoc: () => this.state.doc,
+      getStateUrl: () => this.state.url,
+      registerPreload: (id, element)=>this.registerPreload(id, element),
+      setNeedsLoad: () => {
+        this.needsLoad = true
+      },
+      // eslint-disable-next-line flowtype/no-types-missing-file-annotation
```

Comment:
I'm curious why this would be needed

Reply:
This is part of the reload functionality, now moved to `hv-root`
https://github.com/Instawork/hyperview/blob/dda5288214a223851ca56536df82baaa63d557f9/src/core/components/hv-root/index.tsx#L138-L139


In the original implementation, `hv-screen` set it for itself.
https://github.com/Instawork/hyperview/blob/c6435a7e6f3240b51a131ef6dd7282ae0b93175a/src/core/components/hv-screen/index.js#L283-L284

I didn't want to reach into `hv-screen` from `hv-root` so I provided a callback to allow it to set its value.

Reply:
I meant the eslint comment that disable flow annotation - given the file is not/shouldn't be covered by flow

Reply:
Ha, yeah, I removed that in the next commits. I didn't need to define a type there.